torch/_dynamo/variables/builder.py

```diff
@@ -406,6 +416,10 @@ def _wrap(self, value):
                 # to check for module property mutations, which does a reasonable,
                 # but not completely secure job ensuring a property wasn't changed.
                 guards = self.make_guards(GuardBuilder.BOOL_FALSE)
+            elif keys_are_types_or_namedtuple:
+                # For keys that are types, we only guard on the wheather the dictionary
+                # was modified by checking its version.
+                guards = self.make_guards(GuardBuilder.DICT_VERSION)
```

Comment:
You need an id test too then, right?

Reply:
(In fact, DICT_VERSION should probably imply the id test.)

Reply:
Agreed.